providers/base/units/graphics/jobs.pxu

```diff
@@ -273,19 +273,24 @@ user: root
 
 unit: template
 template-resource: graphics_card
-template-filter: graphics_card.driver in ['amdgpu', 'amdgpu-pro']
+template-filter: graphics_card.driver in ['nvidia', 'pcieport', 'amdgpu', 'amdgpu-pro']
 plugin: shell
 category_id: com.canonical.plainbox::graphics
 id: graphics/{index}_valid_opengl_renderer_{product_slug}
 requires: executable.name == 'glxinfo'
 command:
- renderer=$(DRI_PRIME=1 glxinfo | grep "OpenGL re")
+ # shellcheck disable=SC2050
+ if [[ "{driver}" == "nvidia" || "{driver}" == "pcieport" ]]; then
+     renderer=$(__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia glxinfo | grep "OpenGL re")
+ else
+     renderer=$(DRI_PRIME=1 glxinfo | grep "OpenGL re")
+ fi
```

Comment:
In many graphics jobs (for example [this one](https://github.com/canonical/checkbox/blob/main/providers/base/units/graphics/jobs.pxu#L190)), the first line of the `command` is

```
source graphics_env.sh {driver} {index}
```

If I remember correclty, this allows to set all the required environment variables before calling anything else.

 Is there a specific reason not to use the same mechanism?

Reply:
The `graphics_env.sh` set environment variables while GPUs are more than 1 only.  If you agree the test plan will never set index less than two, it would be great to use the same mechanism.